doc/changelog.md

```diff
@@ -143,6 +143,24 @@
       return qml.expval(qml.Hamiltonian(coeffs, obs))
   ```
 
+* Add support for nested Hamiltonian observables.
+  [#255](https://github.com/PennyLaneAI/catalyst/pull/255)
+
+  ``` python
+  @qjit
+  @qml.qnode(qml.device("lightning.qubit", wires=3))
+  def circuit(x, y, coeffs1, coeffs2):
+      qml.RX(x, wires=0)
+      qml.RX(y, wires=1)
+      qml.RY(x + y, wires=2)
+
+      obs = [
+          qml.PauliX(0) @ qml.PauliZ(1),
+          qml.Hamiltonian(coeffs1, [qml.PauliZ(0) @ qml.Hadamard(2)]),
+      ]
+
+      return qml.var(qml.Hamiltonian(coeffs2, obs))
+  ```

```

Comment:
Quick question @maliasadi -- does this feature also work for the new op arithmetic you are working on in the other open PR?

Reply:
Yes @josh146. This works for the new arithmetic operators and it's basically the last piece of the puzzle. After merging this PR, we can merge PR #250.

Reply:
Yes @josh146. This works for the new operators too and it's technically the last piece of the puzzle. After merging this PR, we should be able to merge #250.